lib/reversetunnel/localsite.go

```diff
@@ -262,6 +242,47 @@ func (s *localSite) dialWithAgent(params DialParams) (net.Conn, error) {
 	return conn, nil
 }
 
+// dialTunnel connects to the target host through a tunnel.
+func (s *localSite) dialTunnel(params DialParams) (net.Conn, error) {
+	rconn, err := s.getRemoteConn(params.ServerID)
+	if err != nil {
+		return nil, trace.NotFound("no tunnel connection found: %v", err)
+	}
+
+	s.log.Debugf("Tunnel dialing to %v.", params.ServerID)
+
+	conn, err := s.chanTransportConn(rconn)
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+
+	return conn, nil
+}
+
+func (s *localSite) getConn(params DialParams) (conn net.Conn, useTunnel bool, err error) {
```

Comment:
@russjones Charn, instead of third parameter, please refactor this method to be more like this:

https://github.com/gravitational/teleport/commit/e3ca4df5fc1efb45e42809463b065fae5b76ddc8#diff-32cceab2787108fd6eadc9b8c982f4ff

Note that in my commit I had to remove the 'useTunnel` everywhere as it cluttered the code paths. Instead I made the 'useTunnel` to be a property of the connection to be used when necessary.

https://github.com/gravitational/teleport/commit/e3ca4df5fc1efb45e42809463b065fae5b76ddc8#diff-b2616f8fc5d7121127fe898e501d328dR176

@russjones please ack.